pygmt/tests/test_surface.py

```diff
@@ -103,13 +103,13 @@ def test_surface_with_outfile_param(ship_data):
 def test_surface_short_aliases(ship_data):
```

Comment:
I recommend adding a test that checks that the function still works and a warning about deprecated syntax is raised when a user supplies an argument to the `outfile` parameter. In the suggested test below, this is done using the same checks as in the `test_surface_with_outgrid_param` test with the addition of `pytest.warns` in a context manager for checking that one warning is raised.
```suggestion
def test_surface_deprecate_outfile_to_outgrid(ship_data):
    """
    Make sure that the old parameter "outfile" is supported and it reports a
    warning.
    """
    with pytest.warns(expected_warning=FutureWarning) as record:
        data = ship_data.values  # convert pandas.DataFrame to numpy.ndarray
        try:
            output = surface(
                data=data, spacing="5m", region=[245, 255, 20, 30], outfile=TEMP_GRID
            )
            assert output is None  # check that output is None since outfile is set
            assert os.path.exists(path=TEMP_GRID)  # check that file exists at path

            with xr.open_dataarray(TEMP_GRID) as grid:
                assert isinstance(grid, xr.DataArray)  # ensure netcdf grid loads ok
        finally:
            os.remove(path=TEMP_GRID)
        assert len(record) == 1 # check that only one warning was raised


def test_surface_short_aliases(ship_data):
```